src/box/memtx_hash.cc

```diff
@@ -443,7 +443,17 @@ memtx_hash_index_create_iterator(struct index *base, enum iterator_type type,
 	light_index_iterator_begin(&index->hash_table, &it->iterator);
 
 	switch (type) {
-	case ITER_GT:
+	case ITER_GT: {
+		static bool warn_once = false;
+		if (!warn_once) {
+			warn_once = true;
+			say_warn("HASH index 'GT' iterator type is deprecated "
+				 "since Tarantool 2.11 and should not be "
+				 "used.\n"
+				 "It will be removed in a future Tarantool "
+				 "release.");
```

Comment:
A few minor comments:
 - Multi-line strings don't look good in the log.
 - I think "It will be removed in a future release" sentence is redundant - it follows from the fact that the feature was deprecated.
 - Maybe mention that 'ALL' should be used instead?
 - Should we maybe backport this to 2.10? The ticket lacks any labels and isn't included into the team plan.

Reply:
> Should we maybe backport this to 2.10?

Since the warning states `since Tarantool 2.11`, I think we shouldn't.

Reply:
Maybe, deprecate it in 2.10 *and* fix the log message :) Need to ask @alyapunov just in case.